R/utils_render_table_venues.R

```diff
@@ -96,41 +126,47 @@ render_table_all_venues_html <- function(list_venue_reg_tables){
 
   # Column- Venue type
   table_venues[[col_names[["venue_type"]]]] <- sapply(
-    X = list_venue_names,
-    FUN = function(venue_name){
-      venue_type <- determine_venue_category(venue_name)
-      link_venue_subcat <- paste0("https://codecheck.org.uk/register/venues/", venue_type, "/")
-      paste0("[", stringr::str_to_title(venue_type), "](", link_venue_subcat,")")
+    seq_along(list_venue_names),
+    FUN = function(i){
+      venue_type <- list_venue_types[[i]]
+      hyperlink <- paste0("https://codecheck.org.uk/register/venues/", venue_type, "/")
+      paste0("[", stringr::str_to_title(venue_type), "](", hyperlink,")")
     }
   )
 
   # Column- No. of codechecks
   table_venues[[col_names[["no_codechecks"]]]] <- sapply(
-    table_venues[[col_names[["venue_name"]]]],
-    FUN = function(venue_name) {
-      venue_type <- determine_venue_category(venue_name)
-      no_codechecks <- nrow(list_venue_reg_tables[[venue_name]])
-      venue_name <-  determine_venue_name(venue_name, venue_type)
-      paste0(no_codechecks," [(see all checks)](https://codecheck.org.uk/register/venues/",
-      venue_type, "/", venue_name, "/)")
+    seq_along(list_venue_names),
+    FUN = function(i) {
+      venue_name <- list_venue_names[[i]]
+      venue_type <- list_venue_types[[i]]
+      no_codechecks <- nrow(list_venue_reg_tables[[venue_type]][[venue_name]])
+
+      # Replacing the " " with "_" for the hyperlinks
+      venue_name <- gsub(" ", "_",venue_name)
+      hyperlink <- paste0("https://codecheck.org.uk/register/venues/", venue_type, "/", venue_name, "/")
+      paste0(no_codechecks, " [(see all checks)](", hyperlink, ")")
     }
   )
 
   # Column- venue names
   # Each venue name will be a hyperlink to the register table
   # with all their codechecks
   table_venues[[col_names[["venue_name"]]]] <- sapply(
-    table_venues[[col_names[["venue_name"]]]],
-    FUN = function(venue_name){
+    seq_along(list_venue_names),
+    FUN = function(i){
+      venue_name <- list_venue_names[[i]]
+      venue_type <- list_venue_types[[i]]
       display_venue_name <- CONFIG$DICT_VENUE_NAMES[[venue_name]]
-
+      
       if (is.null(display_venue_name)) {
         return(venue_name)  # Handle cases where venue_name is not in CONFIG$DICT_VENUE_NAMES
       }
-      venue_type <- determine_venue_category(venue_name)
-      venue_name <-  determine_venue_name(venue_name, venue_type)
-      paste0("[", display_venue_name, "](https://codecheck.org.uk/register/venues/",
-            venue_type, "/", venue_name, "/)")
+      
+      # Replacing the " " with "_" for the hyperlinks
+      venue_name <- gsub(" ", "_",venue_name)
+      hyperlink <- paste0("https://codecheck.org.uk/register/venues/",venue_type, "/", venue_name, "/")
```

Comment:
Please move this constant string to `config.R`.

Reply:
I plan to move all the links to config.R in a separate PR. Is that okay with you?

Reply:
Yes!